mentoring-app/src/app/courses-page/course/course.component.ts

```diff
@@ -40,15 +34,17 @@ export class CourseComponent implements OnInit, OnChanges, AfterContentInit, OnD
   }
 
   public handleDelete(id: number): void {
-    console.log('Course id', id);
+    if (confirm('You really want to delete this course?')) {
+      this.coursesService.deleteCourse(id);
+    }
   }
 
   public onLoadMore(): void {
     console.log('load more');
   }
 
   public onFindClick(): void {
-  this.coursesCopy = this.searchPipe.transform(this.courses, this.searchInput);
+  this.courses = this.searchPipe.transform(this.coursesService.getCourses(), this.searchInput);
```

Comment:
getCourses method is called on every key stroke here

Reply:
fixed